### ISSUE TITLE
Unable to use suggested CommitDocumentedChanges Method

### ISSUE THREAD

**Issue by MMasey** [umbraco/Umbraco-CMS#14112]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.4.2

### Bug summary

I've been trying to automatically set a value in a ContentPublishingNotification, but when I attempt to use the `_contentService.SaveAndPublish(item)` method, it throws an exception saying the following.

`Cannot save-and-publish (un)publishing content, use the dedicated CommitDocumentedChanges method.`. [The line of code is at this link](https://github.com/umbraco/Umbraco-CMS/blob/36069aa8139b6125398786d378ac712b5382d0b0/src/Umbraco.Core/Services/ContentService.cs#L1117)

Unfortunately, the suggested `CommitDocumentedChanges` method is currently set to an internal only method. I'm not sure if this is intentional or not, but at the moment it means the following.

1. It's not possible to amend a value whilst publishing
2. The `node.SetValue(propertyAlias, value);` method is used in a `ContentPublishingNotification` the value will only be saves, but the publish will continue, but it the backoffice is will show the property as being published when it has only been saved.
3. Either the `CommitDocumentedChanges` needs to be public, or the exception needs to be updated with correct messaging.

### Steps to reproduce

1. Create a simple document type with a text string property called "Example Property", and create view capable of rendering out the published property value.
2. Create a ContentPublishingNotification similar to this, and add it via the UmbracoBuilder
```
internal class ExamplePublishedNotification : INotificationHandler<ContentPublishingNotification>
    {
        private readonly IContentService _contentService;

        public ExamplePublishedNotification(
            IContentService contentService)
        {
            _contentService = contentService;
        }

        public void Handle(ContentPublishingNotification notification)
        {
            const string alias = "exampleProperty";

            foreach (var item in notification.PublishedEntities)
            {
                try
                {
                    var dateValue = DateTime.Now;
                    item.SetValue(alias, dateValue.ToString());
                    
                    // This will throw and error, remove it to only save the value and not publish it.
                    _contentService.SaveAndPublish(item); 
                    
                    notification.Messages.Add(new EventMessage(
                        "The \"Example Property\" value has automatically been set to",
                        dateValue.ToString("yyyy-MM-dd"),
                        EventMessageType.Success));
                }
                catch (Exception ex)
                {
                    notification.Messages.Add(new EventMessage(
                        "There was a problem when trying to set the \"Example Property\" value automatically.", null,
                        EventMessageType.Error));
                }
            }
        }
    }
```
3. Test save and publishing the value and you'll see the it only saves it, or errors completely.

### Expected result / actual result

It should be possible to Save and Publish a value via `ContentPublishingNotification` using `_contentService.CommitDocumentedChanges` or another appropriate method.

**Comment by elit0451** [comment]
Hi @MMasey 👋 

Thanks for bringing this to our attention! As you can see in the explanation of #14114, making use of the `ContentPublishingNotification` for your intended scenario is not the preferred way and in the testing notes I've added an example of how to achieve what you are looking for using <code>Content<b>Saving</b>Notification</code>. There is an extra check to make sure that we apply the changes when the content item is to be Saved and published. This would be possible once the PR is merged in.

I will close this issue since this is not the intended way to achieve the given scenario. But it helped us figure out another bug, so thanks for reaching out 💪

**Comment by MMasey** [comment]
Hey @elit0451

That looks like exactly what I need! Thank you for confirming the correct way of doing it, and also adding a fix to achieve what I need to do. I'll be sure to keep an eye out for when it's released.

Many thanks,
Mike

**Comment by markadrake** [comment]
We have logic that requires content to be updated only before publishing. 
In our scenario, the values we assign only matter when the content is published.
Our logic is broken with the changes, and we cannot reinstate the necessary functionality. 
I hope that HQ can advise.

**Comment by elit0451** [comment]
Hi @markadrake,
I am sorry to hear that a fix is causing problems in your setup. Could I ask you to create a new issue on the tracker with the steps to reproduce on a clean install 😊 so we keep things separated
